Configure.ps1

```diff
@@ -82,74 +81,75 @@ of the visual studio solution (App.Config and Web.Config) so that they are consi
-    $creds = Login-AzureRmAccount -TenantId $tenantId
 
     # Create the Azure Active Directory Application and it's service principal
     # Note that if, at this point, you get an error: "New-AzureRmADApplication : Your Azure credentials have not been set up or have expired, please run Login-AzureRMAccount to set up your Azure credentials"
     # then you will need to run Clear-AzureProfile (you might have an expired token)
     Write-Host "Creating the service appplication ($serviceAadAppName)"
-    $serviceApplication = New-AzureRmADApplication -DisplayName $serviceAadAppName -HomePage $serviceHomePage -IdentifierUris $serviceAppIdIRI
-    $serviceservicePrincipal = New-AzureRmADServicePrincipal -ApplicationId $serviceApplication.ApplicationId
+    $serviceApplication =New-AzureADApplication -DisplayName $serviceAadAppName -HomePage $serviceHomePage -IdentifierUris $serviceAppIdURI
+    $serviceservicePrincipal = New-AzureADServicePrincipal -AppId $serviceApplication.AppId
 
     # Create the daemon application
     # ------------------------------
+    # And create the daemon application
+    $daemonApplication = New-AzureADApplication  -DisplayName $daemonAadAppName -HomePage $daemonHomePage -IdentifierUris $daemonAppIdURI
+    $daemonservicePrincipal = New-AzureADServicePrincipal  -AppId $daemonApplication.AppId
+
     # Generate a certificate
     Write-Host "Creating the client appplication ($daemonAadAppName)"
     $certificate=New-SelfSignedCertificate -Subject $certificateName -CertStoreLocation "Cert:\CurrentUser\My"  -KeyExportPolicy Exportable -KeySpec Signature 
     $certKeyId = [Guid]::NewGuid()
-
-    # Create Azure Key Credentials from the certificate
-    $keyCredential = New-Object Microsoft.Azure.Commands.Resources.Models.ActiveDirectory.PSADKeyCredential
-    $keyCredential.KeyId = $certKeyId
-    $keyCredential.CertValue = [System.Convert]::ToBase64String($certificate.GetRawCertData())
-    $keyCredential.StartDate = $certificate.NotBefore
-    $keyCredential.EndDate= $certificate.NotAfter
-
-    # And create the daemon application
-    $daemonApplication = New-AzureRmADApplication -DisplayName $daemonAadAppName -HomePage $daemonHomePage -IdentifierUris $daemonAppIdIRI -KeyCredentials $keyCredential
-    $daemonservicePrincipal = New-AzureRmADServicePrincipal -ApplicationId $daemonApplication.ApplicationId
+    $certBase64Value = [System.Convert]::ToBase64String($certificate.GetRawCertData())
+    $certBase64Thumbprint = [System.Convert]::ToBase64String($certificate.GetCertHash())
+    
+    # Add a Azure Key Credentials from the certificate for the application
+    $applicationKeyCredentials = New-AzureADApplicationKeyCredential -ObjectId $daemonApplication.ObjectId -CustomKeyIdentifier $certificateName -Type AsymmetricX509Cert -Usage Verify -Value $certBase64Value  -StartDate $certificate.NotBefore -EndDate $certificate.NotAfter
```

Comment:
If you want to make this a bit more readable then you can split the lines using the backtick ` character.

```
$applicationKeyCredentials = New-AzureADApplicationKeyCredential -ObjectId $daemonApplication.ObjectId `
                                                                 -CustomKeyIdentifier $certificateName `
                                                                 -Type AsymmetricX509Cert `
                                                                 -Usage Verify `
                                                                 -Value $certBase64Value  `
                                                                 -StartDate $certificate.NotBefore `
                                                                 -EndDate $certificate.NotAfter
```